### PR TITLE
Add WAGTAILADMIN_COMMENTS_ENABLED setting

### PR DESCRIPTION
**Pull request by jacobtoppm**
Adds a setting to globally disable commenting more easily

### CODE REVIEW

File: docs/reference/settings.rst [hunk 1]
```diff
@@ -200,6 +200,15 @@ If a user has not uploaded a profile picture, Wagtail will look for an avatar li
 
 Changes whether the Submit for Moderation button is displayed in the action menu.
 
+Comments
+========
+
+.. code-block:: python
+
+    WAGTAILADMIN_COMMENTS_ENABLED = False
```
Comment (kaedroho) [inline docs/reference/settings.rst:208]: Think we need something that indicates the default here. A comment like "Disable comments" just above this line will do

File: wagtail/admin/tests/test_edit_handlers.py [hunk 2]
```diff
@@ -1061,13 +1061,34 @@ def setUp(self):
...
+        set_default_page_edit_handlers(Page)
+
     def test_comments_toggle_enabled(self):
         """
         Test that the comments toggle is enabled for a TabbedInterface containing CommentPanel, and disabled otherwise
         """
         self.assertTrue(self.tabbed_interface.show_comments_toggle)
         self.assertFalse(TabbedInterface([ObjectList(self.event_page.content_panels)]).show_comments_toggle)
 
+    @clear_edit_handler(Page)
```
Comment (kaedroho) [inline wagtail/admin/tests/test_edit_handlers.py:1074]: You could make this happen automatically using Django's ``setting_changed`` signal. For example: https://github.com/wagtail/wagtail/blob/main/wagtail/core/utils.py#L290

File: docs/reference/settings.rst (hunk 1 above)
Comment (jacobtoppm) [reply docs/reference/settings.rst:208]: Good idea, will add that!

File: wagtail/admin/edit_handlers.py [hunk 3]
```diff
@@ -947,6 +954,16 @@ def get_edit_handler(cls):
 Page.get_edit_handler = get_edit_handler
 
 
+@receiver(setting_changed)
+def reset_cache(**kwargs):
```
Comment (kaedroho) [inline wagtail/admin/edit_handlers.py:958]: Could you give this a more descriptive name?

**Review by kaedroho — APPROVED** [review]

File: wagtail/admin/edit_handlers.py (hunk 3 above)
Comment (jacobtoppm) [reply wagtail/admin/edit_handlers.py]: Will do!

**Pull request merged**